tests/db_functions/json/test_json_array.py

```diff
@@ -107,6 +107,45 @@ def test_order_by_nested_key(self):
         )
         self.assertQuerySetEqual(qs, Author.objects.order_by("-alias"))
 
+    def test_null_on_null(self):
+        obj = Author.objects.annotate(
+            arr=JSONArray(F("goes_by"), absent_on_null=False)
+        ).first()
+
+        self.assertEqual(obj.arr, [None])
+
+    @skipIfDBFeature("supports_json_absent_on_null")
+    def test_absent_on_null_not_supported(self):
+        msg = "ABSENT ON NULL is not supported by this database backend."
+        with self.assertRaisesMessage(NotSupportedError, msg):
+            Author.objects.annotate(
+                arr=JSONArray(F("goes_by"), absent_on_null=True)
+            ).first()
+
+    @skipUnlessDBFeature("supports_json_absent_on_null")
+    def test_absent_on_null(self):
+        obj = Author.objects.annotate(
+            arr=JSONArray(F("name"), F("goes_by"), absent_on_null=True)
+        ).first()
+        self.assertEqual(obj.arr, ["Ivan Ivanov"])
+
```

Comment:
I think it might be worth adding a test to ensure the null eliding is performed at the database level and not at the Python level.

What about

```suggestion
    @skipUnlessDBFeature("supports_json_absent_on_null")
    def test_absent_on_null_transform(self):
        obj = Author.objects.annotate(
            arr=JSONArray("goes_by", "name", absent_on_null=True),
            arr_first=F("arr__0"),
        ).first()
        self.assertEqual(obj.arr_first, "Ivan Ivanov")
```